src/main/java/it/gov/pagopa/rtd/ms/rtdmsdecrypter/event/EventHandler.java

```diff
@@ -43,23 +43,45 @@ public Consumer<Message<List<EventGridEvent>>> blobStorageConsumer(DecrypterImpl
-        .filter(e -> "Microsoft.Storage.BlobCreated".equals(e.getEventType()))
-        .map(EventGridEvent::getSubject)
-        .map(BlobApplicationAware::new)
-        .filter(b -> !BlobApplicationAware.Application.NOAPP.equals(b.getApp()))
-        .map(blobRestConnectorImpl::get)
-        .filter(b -> BlobApplicationAware.Status.DOWNLOADED.equals(b.getStatus()))
-        .map(decrypterImpl::decrypt)
-        .filter(b -> BlobApplicationAware.Status.DECRYPTED.equals(b.getStatus()))
-        .flatMap(blobSplitterImpl::split)
-        .filter(b -> BlobApplicationAware.Status.SPLIT.equals(b.getStatus()))
-        .map(blobVerifierImpl::verify)
-        .filter(b -> BlobApplicationAware.Status.VERIFIED.equals(b.getStatus()))
-        .map(b -> isChunkUploadEnabled ? blobRestConnectorImpl.put(b) : b)
-        .filter(b -> BlobApplicationAware.Status.UPLOADED.equals(b.getStatus()))
-        .map(BlobApplicationAware::localCleanup)
-        .filter(b -> BlobApplicationAware.Status.DELETED.equals(b.getStatus()))
-        .collect(Collectors.toList());
+    return message -> {
+      List<BlobApplicationAware> chunks = message.getPayload().stream()
+          .filter(e -> "Microsoft.Storage.BlobCreated".equals(e.getEventType()))
+          .map(EventGridEvent::getSubject)
+          .map(BlobApplicationAware::new)
+          .filter(b -> !BlobApplicationAware.Application.NOAPP.equals(b.getApp()))
+          .map(blobRestConnectorImpl::get)
+          .filter(b -> BlobApplicationAware.Status.DOWNLOADED.equals(b.getStatus()))
+          .map(decrypterImpl::decrypt)
+          .filter(b -> BlobApplicationAware.Status.DECRYPTED.equals(b.getStatus()))
+          .flatMap(blobSplitterImpl::split)
+          .filter(b -> BlobApplicationAware.Status.SPLIT.equals(b.getStatus()))
+          .collect(Collectors.toList());
+
+      List<BlobApplicationAware> verifiedChunks = chunks.stream()
+          .map(blobVerifierImpl::verify)
+          .filter(b -> BlobApplicationAware.Status.VERIFIED.equals(b.getStatus()))
+          .collect(Collectors.toList());
+
+      if (!chunks.isEmpty()) {
+
+        if (verifiedChunks.size() == chunks.size()) {
+          List<BlobApplicationAware> uploadedChunks = verifiedChunks.stream()
+              .map(b -> isChunkUploadEnabled ? blobRestConnectorImpl.put(b) : b)
+              .filter(b -> BlobApplicationAware.Status.UPLOADED.equals(b.getStatus()))
+              .collect(Collectors.toList());
+          log.info("Uploaded chunks: {}", uploadedChunks.size());
+        } else {
+          log.error("Not all chunks are verified, no chunks will be uploaded");
+        }
+
+        List<BlobApplicationAware> deletedChunks = chunks.stream()
+            .map(BlobApplicationAware::localCleanup)
+            .filter(b -> BlobApplicationAware.Status.DELETED.equals(b.getStatus()))
+            .collect(Collectors.toList());
+
+        log.info("Handled blob: {}", deletedChunks.get(0).getOriginalBlobName());
```

Comment:
`.get(0)` can lead into a NullPointerException, are you sure the list is never empty? If not, you can use a `deletedChunks.findFirst()` to get an optional